src/Linters/PackagePrivateLinter.hack

```diff
@@ -0,0 +1,318 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use type Facebook\HHAST\{
+  ASTLintError,
+  ASTLinter,
+  FunctionToken,
+  INamespaceUseDeclaration,
+  NameToken,
+  NamespaceDeclaration,
+  NamespaceGroupUseDeclaration,
+  NamespaceToken,
+  NamespaceUseClause,
+  NamespaceUseDeclaration,
+  Node,
+  QualifiedName,
+  Script,
+  TypeToken,
+};
+use namespace HH\Lib\{C, Str, Vec};
+
+type use_definition_parts_t = shape(
+  'parts' => vec<string>,
+  'alias' => ?string,
+  'node' => Node,
+);
+type package_private_use_definitions_t = shape(
+  'function' => vec<use_definition_parts_t>,
+  'namespace' => dict<string, use_definition_parts_t>,
+  'type' => vec<use_definition_parts_t>,
+);
+
+final class PackagePrivateLinter extends ASTLinter {
+  const type TContext = Script;
+  const type TNode = Script;
+
+  <<__Override>>
+  public function getLintErrorForNode(
+    this::TContext $context,
+    this::TNode $node,
+  ): ?ASTLintError {
+    // Get the current namespace for the file.
+    $current_namespace_for_file = vec[];
+    $top_level_namespace = $this->getTopLevelNamespace($node);
+    if ($top_level_namespace is nonnull) {
+      $current_namespace_for_file = Str\split($top_level_namespace, '\\');
+    }
+
+    // Process all use statements present in the file.
+    $use_definition_context_shape = $this->getUseDefinitions($context);
+    $use_definition_context = Shapes::toArray(
+      $use_definition_context_shape,
+    );
+
+    // Go through use definition to check for potential violations of package private.
+    foreach ($use_definition_context as $_key => $use_definition_context_value) {
```

Comment:
nit: I'd skip the `$_key =>` entirely - if it's unused and you don't need to specify it, don't, especially when the nameisn't descriptive